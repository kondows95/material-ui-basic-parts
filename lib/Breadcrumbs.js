var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { Box, Link, Breadcrumbs } from '@material-ui/core';
import { NavigateNext as NavigateNextIcon } from '@material-ui/icons';
import { withRouter } from 'react-router-dom';
import { useIntl } from 'react-intl';
var MyBreadcrumbs = function (props) {
    var formatMessage = useIntl().formatMessage;
    var fontSize = props.fontSize || 14;
    var handleClick = function (href) { return function (event) {
        event.preventDefault();
        props.history.push(href);
    }; };
    var MyBox = function (props) { return (React.createElement(Box, __assign({ fontSize: fontSize }, props), props.children)); };
    return (React.createElement(Box, { m: 2 },
        React.createElement(Breadcrumbs, { separator: React.createElement(NavigateNextIcon, { fontSize: "small" }), "aria-label": "breadcrumb" }, props.links.map(function (obj, idx) {
            var label = formatMessage({ id: obj.intlID, defaultMessage: obj.intlID });
            if (obj.href) {
                return (React.createElement(Link, { "data-testid": 'breadcrumb' + idx, key: idx, color: "inherit", onClick: handleClick(obj.href) },
                    React.createElement(MyBox, null, label)));
            }
            else {
                return (React.createElement(MyBox, { "data-testid": 'breadcrumb' + idx, key: idx, color: "text.primary" }, label));
            }
        }))));
};
export default withRouter(MyBreadcrumbs);
