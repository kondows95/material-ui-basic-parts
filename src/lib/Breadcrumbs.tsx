import React from 'react';
import { Box, Link, Breadcrumbs, BoxProps } from '@material-ui/core';
import { NavigateNext as NavigateNextIcon } from '@material-ui/icons';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { useIntl } from 'react-intl';

export type Link = {
    intlID: string;
    href?: string;
};

type Props = {
    links: Link[];
    fontSize?: number | string;
} & RouteComponentProps;

const MyBreadcrumbs: React.FC<Props> = (props) => {
    const { formatMessage } = useIntl();

    const fontSize = props.fontSize || 14;

    const handleClick = (href: string) => (event: React.MouseEvent): void => {
        event.preventDefault();
        props.history.push(href);
    };

    const MyBox = (props: BoxProps): React.ReactElement => (
        <Box fontSize={fontSize} {...props}>
            {props.children}
        </Box>
    );

    return (
        <Box m={2}>
            <Breadcrumbs separator={<NavigateNextIcon fontSize="small" />} aria-label="breadcrumb">
                {props.links.map((obj, idx) => {
                    const label = formatMessage({ id: obj.intlID, defaultMessage: obj.intlID });
                    if (obj.href) {
                        return (
                            <Link
                                data-testid={'breadcrumb' + idx}
                                key={idx}
                                color="inherit"
                                onClick={handleClick(obj.href)}
                            >
                                <MyBox>{label}</MyBox>
                            </Link>
                        );
                    } else {
                        return (
                            <MyBox data-testid={'breadcrumb' + idx} key={idx} color="text.primary">
                                {label}
                            </MyBox>
                        );
                    }
                })}
            </Breadcrumbs>
        </Box>
    );
};

export default withRouter(MyBreadcrumbs);
