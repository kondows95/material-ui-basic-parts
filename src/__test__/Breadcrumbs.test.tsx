import React from 'react';
import TestLib from 'react-testing-library-wrapper';
import { IntlProvider } from 'react-intl';
import { BrowserRouter } from 'react-router-dom';
import Breadcrumbs from '../Breadcrumbs';

let tLib: TestLib;

const links = [{ intlID: 'home', href: '/' }, { intlID: 'category1', href: '/category1' }, { intlID: 'article1' }];

describe('Basic Usage', () => {
    beforeEach(() => {
        const langMessages = { home: 'HOME', category1: 'CATEGORY1', article1: 'ARTICLE1' };
        tLib = new TestLib(
            (
                <IntlProvider locale="en" messages={langMessages}>
                    <BrowserRouter>
                        <Breadcrumbs links={links} />
                    </BrowserRouter>
                </IntlProvider>
            )
        );
    });

    it('You can set TOP page link and label', () => {
        tLib.click('breadcrumb0');
        expect(location.pathname).toBe('/');
        const elem = tLib.get('breadcrumb0');
        expect(elem.textContent).toBe('HOME');
        expect(elem.tagName).toBe('A');
    });

    it('You can set PARENT page link and label', () => {
        tLib.click('breadcrumb1');
        expect(location.pathname).toBe('/category1');
        const elem = tLib.get('breadcrumb1');
        expect(elem.textContent).toBe('CATEGORY1');
        expect(elem.tagName).toBe('A');
    });

    it('Current page is not link', () => {
        const elem = tLib.get('breadcrumb2');
        expect(elem.textContent).toBe('ARTICLE1');
        expect(elem.tagName).toBe('DIV');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});

describe('Expansion Usage', () => {
    beforeEach(() => {
        tLib = new TestLib(
            (
                <IntlProvider locale="en">
                    <BrowserRouter>
                        <Breadcrumbs links={links} />
                    </BrowserRouter>
                </IntlProvider>
            )
        );
    });

    it('If Intl messages undefined, label be intlID', () => {
        expect(tLib.get('breadcrumb0').textContent).toBe('home');
        expect(tLib.get('breadcrumb1').textContent).toBe('category1');
        expect(tLib.get('breadcrumb2').textContent).toBe('article1');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});
