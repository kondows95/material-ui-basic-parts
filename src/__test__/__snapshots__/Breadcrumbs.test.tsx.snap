// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Usage Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-170"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      <ol
        class="MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            data-testid="breadcrumb0"
          >
            <div
              class="MuiBox-root MuiBox-root-211"
            >
              HOME
            </div>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            data-testid="breadcrumb1"
          >
            <div
              class="MuiBox-root MuiBox-root-221"
            >
              CATEGORY1
            </div>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <div
            class="MuiBox-root MuiBox-root-222"
            data-testid="breadcrumb2"
          >
            ARTICLE1
          </div>
        </li>
      </ol>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Expansion Usage Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-278"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      <ol
        class="MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            data-testid="breadcrumb0"
          >
            <div
              class="MuiBox-root MuiBox-root-319"
            >
              home
            </div>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            data-testid="breadcrumb1"
          >
            <div
              class="MuiBox-root MuiBox-root-329"
            >
              category1
            </div>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <div
            class="MuiBox-root MuiBox-root-330"
            data-testid="breadcrumb2"
          >
            article1
          </div>
        </li>
      </ol>
    </nav>
  </div>
</DocumentFragment>
`;
